contracts/src/types/Predicate.sol

```diff
@@ -5,7 +5,10 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.24;
 
+import {ReceiptClaim, ReceiptClaimLib} from "risc0/IRiscZeroVerifier.sol";
+
 using PredicateLibrary for Predicate global;
+using ReceiptClaimLib for ReceiptClaim;
 
 /// @title Predicate Struct and Library
 /// @notice Represents a predicate and provides functions to create and evaluate predicates.
```

Comment:
We are now overloading a bit the semantic of the data field here, so might be worth expanding the description explaining what data might contain based on the different types?